File: CPP/Tendik.cpp
//NIP, Jabatan, UnitKerja
#pragma once
#include <iostream>
#include <string>
#include "Manusia.cpp"
#include "UnitKerja.cpp"

using namespace std;

class Tendik : public Manusia{
    private:
        string NIP;
        string Jabatan;
        UnitKerja Unit;
    
    public:
        Tendik(){
            setNIK("");
            setNama("");
            setJenisKelamin("");
            NIP = "";
            Jabatan = "";
        };
        Tendik(string nik, string nama, string jk, string nip, string jabatan, UnitKerja unit){
            setNIK(nik);
            setNama(nama);
            setJenisKelamin(jk);
            NIP = nip;
            Jabatan = jabatan;
            Unit = unit;
        }

        //setter
        void setNIP(string nik){
            NIP = nik;
        }
        void setJabatan(string nik){
            Jabatan = nik;
        }
        void setUnit(UnitKerja nik){
            Unit = nik;
        }

        //getter
        string getNIP(){
            return NIP;
        }
        string getJabatan(){
            return Jabatan;
        }
        UnitKerja getUnit(){
            return Unit;
        }
        ~Tendik(){
        }
};

File: JAVA/Tendik.java
//NIP, Jabatan, UnitKerja

public class Tendik extends Manusia{
    private String NIP;
    private String Jabatan;
    private UnitKerja Unit;

    public Tendik(){
        setNIK("");
        setNama("");
        setJenisKelamin("");
        this.NIP = "";
        this.Jabatan = "";
    }

    public Tendik(String nik, String nama, String jk, String nip, String jabatan, UnitKerja unit){
        setNIK(nik);
        setNama(nama);
        setJenisKelamin(jk);
        this.NIP = nip;
        this.Jabatan = jabatan;
        this.Unit = unit;
    }

    // Setter
    public void setNIP(String nik){
        this.NIP = nik;
    }

    public void setJabatan(String nik){
        this.Jabatan = nik;
    }

    public void setUnit(UnitKerja nik){
        this.Unit = nik;
    }

    // Getter
    public String getNIP(){
        return this.NIP;
    }

    public String getJabatan(){
        return this.Jabatan;
    }

    public UnitKerja getUnit(){
        return this.Unit;
    }
}